slack-code-runner-bot/function.js

```diff
@@ -0,0 +1,123 @@
+const { WebClient } = require('@slack/client');
+const { VM } = require('vm2');
+const decode = require('decode-html');
+
+const {
+  SLACK_BOT_TOKEN: token,
+  BOT_TRIGGER_STATEMENT: triggerStatement,
+  MAX_CODE_RUNTIME_MS: maxTimeMS,
+} = process.env;
+const web = new WebClient(token);
+
+const triBacktick = '```';
+
+/**
+ * Extracts the potential JavaScript code segment located
+ * between leading and trailing triple backticks ```
+ *
+ * @param {string} textToExtractFrom - segment to attempt code extraction on
+ *
+ * @return {string} - extracted code segment
+ */
+function extractBetweenBackticks(textToExtractFrom) {
+  const startIdx = textToExtractFrom.indexOf(triBacktick);
+  const endIdx = textToExtractFrom.lastIndexOf(triBacktick);
+
+  if (startIdx === -1 || endIdx === -1 || (startIdx === endIdx)) {
+    throw new Error('Javascript input must begin and end with triple backticks');
```

Comment:
1. Nit: "JavaScript".
  a. Meganit: "ECMAScript" :-/
2. The error message is unclear: the message doesn't have to start and end with triple backticks.  Maybe something like "No code to run found between triple backticks"?

Reply:
1. Will fix
2. I don't understand completely. If there is not two sets of ``` at the start and end of the text this case is hit. In fact I don't even handle the case of empty '``````'.